### ISSUE TITLE
MissingSuperCall never reports

### ISSUE THREAD

**Issue by mgroth0** [detekt/detekt#7582]
## Expected Behavior

MissingSuperCall will report when the super call is missing.

## Observed Behavior

I can not get the rule to ever report.

## Steps to Reproduce

1. activate the MissingSuperCall rule with its default configuration
2. Include this code in a kotlin/jvm module:

```kotlin
open class ASuperClass {
    @OverridingMethodsMustInvokeSuper
    open fun superFun() = Unit
}
class BreaksDetektRule: ASuperClass() {
    override fun superFun() {
    }
}
```

3. Run detekt
4. Observe no issues are reported

## Your Environment

* Version of detekt used: Commit 7675ae87250594f16184069f7ac2c300d86ed070
* Version of Gradle used (if applicable): 8.8

**Comment by cortinico** [comment]
Are you running with Types Resolution?

**Comment by BraisGabin** [comment]
> Are you running with Types Resolution?

Can you answer this? How are you executing detekt? Which detekt task are you running?

**Comment by atulgpt** [comment]
Hi, @mgroth0 there is already one TC(`super method has OverridingMethodsMustInvokeSuper annotation`) similar to you case. Did you add somehow override `mustInvokeSuperAnnotations` in `MissingSuperCall` rule where `javax.annotation.OverridingMethodsMustInvokeSuper` is set?
Also, can you add the link to your `detekt.yml` from your reproducer project? I tried but couldn't find any

**Comment by atulgpt** [comment]
Hi @mgroth0 ahh I see this is added for 2.0.0 branch so it will not be available for the previous stable release.

> I don't know if detekt publishes stable development builds
This rule will only be stable with the releae of `2.0.0`

But since this rule is disabled by default you need to enable it. Also, you need to add the config like below

```yml
MissingSuperCall:
    active: true
    mustInvokeSuperAnnotations: ['androidx.annotation.CallSuper', 'javax.annotation.OverridingMethodsMustInvokeSuper']
```

`mustInvokeSuperAnnotations` is used to configure the annotations where you want this rule to be invoked

Your msg indicates that since detekt is not running with the type resolution this rule is being skipped.

You `detektAll` task which I can see in your project is defined like below

```groovy
tasks.register("allDetekt") {
    dependsOn(tasks.withType<Detekt>())
}
```

Since it depends on `Detekt` I think it will not be depending on the type resolution version.
You can update your task to like below(syntax might be wrong please cross check):

```groovy
tasks.register("allDetekt") {
    dependsOn(tasks.named("detektDebug")) // Here I am using `detektDebug` which is using type resolution
}
```

**Comment by atulgpt** [comment]
Hi, @mgroth0 I cloned your repo and somehow I am not able to run even a single rule(I added other issues as well in the code but none got reported). So I am not sure if this issue with your project or some configuration

**Comment by atulgpt** [comment]
Hi, @mgroth0 updated the original comment. In the project when you run detekt task, is detekt able to report any issue?

**Comment by atulgpt** [comment]
Hi, @mgroth0 sorry for the late reply, I got the chance to look into the newest reproducer and your added violation(empty catch) is getting reported by the detekt. I am unsure why I could not run my enabled rules in your project.

But I added the below code in your project
```kotlin
open class ASuperClass {
    @OverridingMethodsMustInvokeSuper
    open fun superFun() = Unit
}

class BreaksDetektRule : ASuperClass() {
    override fun superFun() {
    }
}

annotation class OverridingMethodsMustInvokeSuper
```

Just after the line 20 in your `MyComposeAndroidActivity.kt` and added `mustInvokeSuperAnnotations: ['androidx.annotation.CallSuper', 'javax.annotation.OverridingMethodsMustInvokeSuper', 'bug.OverridingMethodsMustInvokeSuper', 'android.annotation.CallSuper']` in the `MissingSuperCall` rule like below
```yml
potential-bugs:
  active: true
  MissingSuperCall:
    active: true
    mustInvokeSuperAnnotations: ['androidx.annotation.CallSuper', 'javax.annotation.OverridingMethodsMustInvokeSuper', 'bug.OverridingMethodsMustInvokeSuper', 'android.annotation.CallSuper']
```
and I am able to get the violation by detekt
```txt
potential-bugs
	MissingSuperCall - [Overriding method is missing a call to overridden super method.] at /Users/atulgup/Learning/open_source/detekt-7582/src/main/kotlin/bug/MyComposeAndroidActivity.kt:28:5
```

I hypothesise that since you are using `androidx.activity.ComponentActivity` which has base class of `android.app.Activity` with the annotation `android.annotation.CallSuper` on `onDestroy` method, this was not getting detected. Here `android.annotation.CallSuper` is unrecognised in the project(so in the actual analysis this will not be resolved). 
Can you try this with androidx activity?

**Comment by atulgpt** [comment]
`So this leads me to conclude that MissingSuperCall might not work if the super class is from a dependency.` For proving this, we have to create a sample project where we can replicate the same. Actually, I think in the Activity class the actual `CallSuper` is not added to your project. These types of classes are classified as NonExistence class from the kotlin compiler analysis

**Comment by BraisGabin** [comment]
OK, closing this one in favour of the other issue. If you think this should be still open please comment here.
